### PR TITLE
[GLUTEN-6827][VL] Add a new test case for Round's coverage

### PR DESCRIPTION
**Pull request by jiangjiangtian**
A new test case found in https://github.com/apache/incubator-gluten/issues/6827.
After fix, we can uncomment it.

### CODE REVIEW

**Comment by jiangjiangtian** [comment]
cc @zhztheplayer

**Comment by zhztheplayer** [comment]
Hi @jiangjiangtian, would you think it's possible to add a case to `TestOperator.scala` as well? Which is using SQL and have result comparison with vanilla Spark in it. So we make it's clear that Gluten gives different answer for the case.

**Comment by jiangjiangtian** [comment]
> Hi @jiangjiangtian, would you think it's possible to add a case to `TestOperator.scala` as well? Which is using SQL and have result comparison with vanilla Spark in it. So we make it's clear that Gluten gives different answer for the case.

@zhztheplayer Thank for your remind. I have add a test in `TestOperator.scala`. Please review.

File: backends-velox/src/test/scala/org/apache/gluten/execution/TestOperator.scala [hunk 1]
```diff
@@ -2078,4 +2078,12 @@ class TestOperator extends VeloxWholeStageTransformerSuite with AdaptiveSparkPla
...
     }
   }
+
+  // Open it after fix
+  // test("Test round expression") {
+  //   val df1 = runQueryAndCompare("SELECT round(cast(0.5549999999999999 as double), 2)") { _ => }
+  //   checkLengthAndPlan(df1, 1)
+  //   val df2 = runQueryAndCompare("SELECT round(cast(0.19324999999999998 as double), 2)") { _ => }
+  //   checkLengthAndPlan(df2, 1)
+  // }
```
Comment (zhztheplayer) [inline backends-velox/src/test/scala/org/apache/gluten/execution/TestOperator.scala:2088]: Comparing to commenting the code out, you can use `ignore` scalatest flag, e.g.,

```scala
ignore("Test round expression") {
  ...
}
```

File: backends-velox/src/test/scala/org/apache/gluten/execution/TestOperator.scala [hunk 2]
```diff
@@ -2078,4 +2078,12 @@ class TestOperator extends VeloxWholeStageTransformerSuite with AdaptiveSparkPla
       checkGlutenOperatorMatch[SortExecTransformer]
     }
   }
+
+  // Open it after fix
```
Comment (zhztheplayer) [inline backends-velox/src/test/scala/org/apache/gluten/execution/TestOperator.scala:2082]: Better change to `// Enable the test after fixing https://github.com/apache/incubator-gluten/issues/6827`, thanks.

File: gluten-ut/spark32/src/test/scala/org/apache/spark/sql/catalyst/expressions/GlutenMathExpressionsSuite.scala [hunk 3]
```diff
@@ -124,6 +124,8 @@ class GlutenMathExpressionsSuite extends MathExpressionsSuite with GlutenTestsTr
     checkEvaluation(Round(1.12345678901234567, 8), 1.12345679)
     checkEvaluation(Round(-0.98765432109876543, 5), -0.98765)
     checkEvaluation(Round(12345.67890123456789, 6), 12345.678901)
+    // Open it after fix
```
Comment (zhztheplayer) [inline gluten-ut/spark32/src/test/scala/org/apache/spark/sql/catalyst/expressions/GlutenMathExpressionsSuite.scala:127]: ditto

File: gluten-ut/spark33/src/test/scala/org/apache/spark/sql/catalyst/expressions/GlutenMathExpressionsSuite.scala [hunk 4]
```diff
@@ -252,6 +252,8 @@ class GlutenMathExpressionsSuite extends MathExpressionsSuite with GlutenTestsTr
     checkEvaluation(Round(1.12345678901234567, 8), 1.12345679)
     checkEvaluation(Round(-0.98765432109876543, 5), -0.98765)
     checkEvaluation(Round(12345.67890123456789, 6), 12345.678901)
+    // Open it after fix
```
Comment (zhztheplayer) [inline gluten-ut/spark33/src/test/scala/org/apache/spark/sql/catalyst/expressions/GlutenMathExpressionsSuite.scala:255]: ditto

File: gluten-ut/spark34/src/test/scala/org/apache/spark/sql/catalyst/expressions/GlutenMathExpressionsSuite.scala [hunk 5]
```diff
@@ -248,6 +248,8 @@ class GlutenMathExpressionsSuite extends MathExpressionsSuite with GlutenTestsTr
     checkEvaluation(BRound(-3.5, 0), -4.0)
     checkEvaluation(BRound(-0.35, 1), -0.4)
     checkEvaluation(BRound(-35, -1), -40)
+    // Open it after fix
```
Comment (zhztheplayer) [inline gluten-ut/spark34/src/test/scala/org/apache/spark/sql/catalyst/expressions/GlutenMathExpressionsSuite.scala:251]: ditto

File: gluten-ut/spark35/src/test/scala/org/apache/spark/sql/catalyst/expressions/GlutenMathExpressionsSuite.scala [hunk 6]
```diff
@@ -251,6 +251,8 @@ class GlutenMathExpressionsSuite extends MathExpressionsSuite with GlutenTestsTr
     checkEvaluation(Round(1.12345678901234567, 8), 1.12345679)
     checkEvaluation(Round(-0.98765432109876543, 5), -0.98765)
     checkEvaluation(Round(12345.67890123456789, 6), 12345.678901)
+    // Open it after fix
```
Comment (zhztheplayer) [inline gluten-ut/spark35/src/test/scala/org/apache/spark/sql/catalyst/expressions/GlutenMathExpressionsSuite.scala:254]: ditto

**Review by zhztheplayer — APPROVED** [review]
👍

**Comment by jiangjiangtian** [comment]
@zhztheplayer Sorry, I forget to add a comment and I add it just now. Please review again.

**Review by zhztheplayer — APPROVED** [review]
Thanks!

**Pull request merged**